presto-hive-common/src/main/java/com/facebook/presto/hive/BlockLocation.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.drift.annotations.ThriftConstructor;
+import com.facebook.drift.annotations.ThriftField;
+import com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class BlockLocation
+{
+    private final List<String> hosts;
+    private final long offset;
+    private final long length;
+
+    @ThriftConstructor
+    public BlockLocation(List<String> hosts, long offset, long length)
+    {
+        this.hosts = requireNonNull(hosts, "hosts is null");
+        this.offset = offset;
+        this.length = length;
+    }
+
+    public BlockLocation(org.apache.hadoop.fs.BlockLocation blockLocation)
+    {
+        requireNonNull(blockLocation, "blockLocation is null");
+        try {
+            this.hosts = ImmutableList.copyOf(blockLocation.getHosts());
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        this.offset = blockLocation.getOffset();
+        this.length = blockLocation.getLength();
+    }
+
+    public static BlockLocation[] fromHiveBlockLocations(@Nullable org.apache.hadoop.fs.BlockLocation[] blockLocations)
+    {
+        if (blockLocations == null) {
+            return new BlockLocation[0];
+        }
+
+        return Arrays.stream(blockLocations)
+                .map(BlockLocation::new)
+                .toArray(BlockLocation[]::new);
+    }
+
+    @ThriftField(1)
+    public List<String> getHosts()
```

Comment:
probably OK, but you are potentially exposing mutable internal state

Reply:
Its an immutable list. And is same as what it is today.